lib/spack/spack/package.py

```diff
@@ -2608,6 +2609,20 @@ def _run_default_install_time_test_callbacks(self):
                 fn()
 
 
+def empty_function(pkg, func):
+    """Returns True if pkg's func is defined as 'pass'; False otherwise."""
+    function = func
+    if isinstance(func, functools.partial):
+        method_self = func.func.__self__
+        if isinstance(method_self, SpecMultiMethod):
+            function = method_self.get_first_matching_method(pkg, None)
+
+    source = (inspect.getsource(function)).splitlines()[1:]
+    lines = (ln.strip() for ln in source)
+    sttmts = [ln for ln in lines if not ln.startswith('#')]
+    return len(sttmts) > 0 and sttmts[0] == 'pass'
```

Comment:
Would it be fine to explicitly mark the function being "empty", instead of parsing the source code? If e.g. we attach an arbitrary boolean attribute to the "empty" functions with a decorator this code can be as simple as using `hasattr`.